src/Symfony/Contracts/Translation/TranslatorTrait.php

```diff
@@ -35,7 +35,7 @@ public function setLocale(string $locale)
      */
     public function getLocale(): string
     {
-        return $this->locale ?: (class_exists(\Locale::class) ? \Locale::getDefault() : 'en');
+        return $this->locale ?? (class_exists(\Locale::class) ? \Locale::getDefault() : 'en');
```

Comment:
this is going to change the outcome for the empty string

Reply:
AFAIR, the component forbids using the empty string as locale. Maybe the setter should forbid the empty string here too.

Reply:
I've initialized the property with `null` for now to keep things simple. If we want to disallow the empty string as @stof suggested, we would need to build a deprecation layer in constracts v2 first.

Reply:
Actually not, see https://github.com/symfony/symfony/pull/42128...